clippy_dev/src/update_lints.rs

```diff
@@ -316,6 +317,168 @@ pub fn rename(old_name: &str, new_name: &str, uplift: bool) {
+        }
+        println!("note: you must run `cargo uitest` to update the test results");
+    }
+
+    let reason = reason.unwrap_or(DEFAULT_DEPRECATION_REASON);
+    let name_lower = name.to_lowercase();
+    let name_upper = name.to_uppercase();
+
+    let (mut lints, deprecated_lints, renamed_lints) = gather_all();
+    let Some(lint) = lints.iter().find(|l| l.name == name_lower) else { panic!("failed to find lint `{}`", name) };
+
+    let mut file_name = file_name.map_or_else(|| format!("{}.rs", name), ToString::to_string);
+    // Lint is in another module (clippy_lints/src/some_mod/some_lint.rs)
+    if lint.module != name.to_lowercase() {
+        file_name = format!("{}/{}", lint.module, file_name);
+    }
+    let clippy_lints_path = clippy_project_root().join(format!("clippy_lints/src/{}", file_name));
+
+    let file_path = Path::new(&clippy_lints_path);
+    let deprecated_lints_path = &*clippy_project_root().join("clippy_lints/src/deprecated_lints.rs");
+
+    if remove_lint_declaration(&name_lower, file_path, &mut lints).unwrap_or(false) {
+        declare_deprecated(&name_upper, deprecated_lints_path, reason).unwrap();
+        finish((lints, deprecated_lints, renamed_lints), name, reason);
+        return;
+    }
+
+    eprintln!("error: lint not found");
+}
+
+fn remove_lint_declaration(name: &str, path: &Path, lints: &mut Vec<Lint>) -> io::Result<bool> {
+    fn remove_lint(name: &str, lints: &mut Vec<Lint>) {
+        lints.iter().position(|l| l.name == name).map(|pos| lints.remove(pos));
+    }
+
+    fn remove_test_assets(name: &str) {
+        let test_file_stem = format!("tests/ui/{}", name);
+        let path = Path::new(&test_file_stem);
+
+        // Some lints have their own directories, delete them
+        if path.is_dir() {
+            fs::remove_dir_all(path).ok();
+            return;
+        }
+
+        // Remove all related test files
+        fs::remove_file(path.with_extension("rs")).ok();
+        fs::remove_file(path.with_extension("stderr")).ok();
+        fs::remove_file(path.with_extension("fixed")).ok();
+    }
+
+    fn remove_impl_lint_pass(lint_name_upper: &str, content: &mut String) {
+        let impl_lint_pass_start = content
+            .find("impl_lint_pass!")
+            .unwrap_or_else(|| panic!("failed to find `impl_lint_pass`"));
+        let mut impl_lint_pass_end = (&content[impl_lint_pass_start..])
+            .find(']')
+            .expect("failed to find `impl_lint_pass` terminator");
+
+        impl_lint_pass_end += impl_lint_pass_start;
+        let stripped = content[impl_lint_pass_start..impl_lint_pass_end]
+            .lines()
+            .filter(|l| !l.contains(lint_name_upper))
+            .fold(String::new(), |s, line| s + line + "\n");
+
+        content.replace_range(impl_lint_pass_start..impl_lint_pass_end, &*stripped);
+    }
+
+    if path.exists() {
+        // We can't delete the entire file, just remove the declaration
```

Comment:
In this case I would emit a message, that code will have to be removed from the module.

Reply:
Good call!